test/registry/TestReverseRegistrar.js

```diff
@@ -127,191 +167,216 @@ describe('ReverseRegistrar Tests', () => {
+    //     assert.equal(await ens.resolver(node), resolver.address)
+    //   })
+
+    //   it('forbids an account with a different address', async () => {
+    //     await exceptions.expectFailure(
+    //       registrar.claimWithResolverForAddr(
+    //         accounts[0],
+    //         accounts[0],
+    //         resolver.address,
+    //         {
+    //           from: accounts[1],
+    //         }
+    //       )
+    //     )
+    //   })
+
+    //   it('allows an owner() of a contract to claimWithResolverForAddr on behalf of the contract', async () => {
+    //     await registrar.claimWithResolverForAddr(
+    //       dummyOwnable.address,
+    //       accounts[0],
+    //       resolver.address,
+    //       {
+    //         from: accounts[0],
+    //       }
+    //     )
+    //     assert.equal(await ens.owner(dummyOwnableReverseNode), accounts[0])
+    //     assert.equal(
+    //       await ens.resolver(dummyOwnableReverseNode),
+    //       resolver.address
+    //     )
+    //   })
+    // })
+
+    describe('setName', () => {
+      it('sets name records', async () => {
+        await registrar.setName('testname', { from: accounts[0], gas: 1000000 })
+        assert.equal(await ens.resolver(node), defaultResolver.address)
+        assert.equal(await defaultResolver.name(node), 'testname')
       })
 
       it('event ReverseClaimed is emitted', async () => {
-        const tx = await registrar.claimWithResolverForAddr(
-          accounts[0],
-          accounts[0],
-          resolver.address,
-          {
-            from: accounts[0],
-          }
-        )
+        const tx = await registrar.setName('testname', {
+          from: accounts[0],
+          gas: 1000000,
+        })
         assertReverseClaimedEventEmitted(tx, accounts[0], node)
       })
+    })
 
-      it('allows an account authorised by the address to specify resolver', async () => {
-        await ens.setApprovalForAll(accounts[1], true, { from: accounts[0] })
-        await registrar.claimWithResolverForAddr(
-          accounts[0],
-          accounts[0],
-          resolver.address,
-          {
-            from: accounts[1],
-          }
-        )
-        assert.equal(await ens.owner(node), accounts[0])
-        assert.equal(await ens.resolver(node), resolver.address)
-      })
-
-      it('allows a controller specify resolver', async () => {
-        await registrar.setController(accounts[1], true)
-        await registrar.claimWithResolverForAddr(
-          accounts[0],
+    describe('setNameForAddr', () => {
+      it('allows controller to set name records for other accounts', async () => {
+        await registrar.setController(accounts[0], true)
+        await registrar.setNameForAddr(
+          accounts[1],
           accounts[0],
           resolver.address,
+          'testname',
           {
-            from: accounts[1],
+            from: accounts[0],
+            gas: 1000000,
```

Comment:
why do you need to set gas at individual test? Can you not adjust at hardhat config or somewhere global config?

Reply:
deleted these. I think they were set to 0 causing tests to fail, so i set it higher, but i think the default should work